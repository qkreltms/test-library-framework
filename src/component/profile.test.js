// https://testing-library.com/docs/guide-which-query
// https://testing-library.com/docs/dom-testing-library/cheatsheet#queries

import React from "react";
import { render } from "@testing-library/react";
import Profile from "./profile";

describe("<Profile />", () => {
  it("matches snapshot", () => {
    // 컴포넌트 렌더링 할때 render함수 사용
    const component = render(<Profile username="username1" name="name1" />);
    // container는 해당 컴포넌트의 최상위 DOM을 가리킴
    expect(component.container).toMatchSnapshot();
  });

  it("shows the props correctly", () => {
    const component = render(<Profile username="username1" name="name1" />);
    component.getByText("username1"); // username1 이라는 텍스트를 가진 엘리먼트가 있는지 확인: <b>{username}</b>&nbsp;
    component.getByText("(name1)");
    component.getAllByText(/na/); // 정규식 /na/를 통과하는 엘리먼트가 있는지 확인
  });
});
