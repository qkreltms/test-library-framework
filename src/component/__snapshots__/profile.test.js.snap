// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div
    data-testid="profile"
  >
    <b>
      username1
    </b>
    Â 
    <span>
      (
      name1
      )
    </span>
  </div>
</div>
`;
